src/Simplex/Chat.hs

```diff
@@ -1724,6 +1733,30 @@ processChatCommand' vr = \case
                 ci <- saveSndChatItem user (CDGroupSnd gInfo) msg (CISndGroupEvent gEvent)
                 toView $ CRNewChatItem user (AChatItem SCTGroup SMDSnd (GroupChat gInfo) ci)
           pure CRMemberRoleUser {user, groupInfo = gInfo, member = m {memberRole = memRole}, fromRole = mRole, toRole = memRole}
+  APIBlockMemberForAll groupId gmId blocked -> withUser $ \user -> do
+    Group gInfo members <- withStore $ \db -> getGroup db vr user groupId
+    let (targetMember, remainingMembers) = partition ((== gmId) . groupMemberId') members
+    case targetMember of
+      [blockedMember] -> do
+        assertUserGroupRole gInfo $ max GRAdmin bmRole
+        withChatLock "blockForAll" . procCmd $
+          if blocked /= isJust blockedByGroupMemberId
+            then do
+              (msg, _) <- sendGroupMessage' user gInfo remainingMembers $ XGrpMemBlock bmMemberId blocked
+              let ciContent = CISndGroupEvent $ SGEMemberBlocked gmId (fromLocalProfile bmp) blocked
+              ci <- saveSndChatItem user (CDGroupSnd gInfo) msg ciContent
+              toView $ CRNewChatItem user (AChatItem SCTGroup SMDSnd (GroupChat gInfo) ci)
+              bm' <- withStore $ \db -> do
+                liftIO $ updateGroupMemberBlockedForAll db user groupId gmId byGMId_
+                getGroupMember db user groupId gmId
+              toggleNtf user bm' (not blocked)
+              pure CRMemberBlockedForAllUser {user, groupInfo = gInfo, member = bm', blocked}
+            else throwChatError $ CECommandError $ if blocked then "already blocked" else "already unblocked"
+        where
+          byGMId_ = if blocked then Just (groupMemberId' membership) else Nothing
+          GroupInfo {membership} = gInfo
+          GroupMember {memberId = bmMemberId, memberRole = bmRole, memberProfile = bmp, blockedByGroupMemberId} = blockedMember
```

Comment:
this seems wrong to overuse this matching in where when it fits well in the case

Reply:
that some long pattern matches can to be done separately, doesn't mean that it has to always be done like that -the default style is to match in place.

Reply:
blockedByGroupMemberId, in particular, can be just used as a predicate to `m`

Reply:
matching in place results in very long lines. moved to `let`s